src/fluree/db/query/exec/eval.cljc

```diff
@@ -117,16 +118,166 @@
   (str/ends-with? s substr))
 
 (defn subStr
-  [s start end]
-  (subs s start end))
+  ([s start]
+   (subStr s start (count s)))
+  ([s start length]
+   ;; The index of the first character in a string is 1.
+   (let [start (dec start)]
+     (subs s start (min (+ start length)
+                        (count s))))))
+
+(defn strLen
+  [s]
+  (count s))
+
+(defn ucase
+  [s]
+  (str/upper-case s))
+
+(defn lcase
+  [s]
+  (str/lower-case s))
+
+(defn contains
+  [s substr]
+  (str/includes? s substr))
+
+(defn strBefore
+  [s substr]
+  (-> (str/split s (re-pattern substr))
+      first
+      str))
```

Comment:
why the extra `str` call here? Also, I don't think this matches the spec. If `substr` doesn't appear in `s`, [the spec says to return an empty literal](https://www.w3.org/TR/sparql11-query/#func-strbefore), but this will return `s`.